src/embedding.rs

```diff
@@ -77,6 +88,81 @@ impl MatrixWrapper for TwoDimVectorMatrix {
         }
     }
 
+    fn init_with_evec(
+        rows: usize,
+        cols: usize,
+        sparse_matrix_reader: Arc<SparseMatrix>,
+        order: TruncatedOrder,
+    ) -> Self {
+        let matrix_transform = |x: ArrayView2<f32>| -> Array2<f32> {
+            let shape = x.shape();
+            let mut arr = Array2::zeros((shape[0], shape[1])); 
+    
+            let sparse_laplacian_entries: Vec<Entry> = sparse_matrix_reader.entries.clone() // Might be a problem?
+            .into_par_iter()
+            .map(|entry| {
+                if entry.col == entry.row { // What happens for reflexive columns?
```

Comment:
>// What happens for reflexive columns?

Could you elaborate more? What is the concern here?

In the end such comments are not suitable in final product code. Its okay for local notes/development but not for final release

Reply:
The concern was twofold. First, I was not sure about how the sparse matrix storing works and was a bit worried about what the case when the cols and the rows are corresponding to the two datasets stored in the sparse matrix, but this is not the case, since both the rows and the cols store both - this is why we get a symmetric matrix.
Also, as one of the later comments pointed out, this code is not working as intended, since there are no diagonal entries in any of the sparse matrices, hence the diagonal entries have to be added in a different loop.
This confusion led to me thinking that in reflexive columns we would possibly overwrite values that are stored on the diagonal, but in the light of the above, this is not relevant.